ext/user/main.php

```diff
@@ -326,6 +331,33 @@ public function onPageRequest(PageRequestEvent $event): void
             $page->set_mode(PageMode::REDIRECT);
             $page->set_redirect(make_link("user/" . $user->name));
         }
+
+        if ($event->page_matches("set_avatar/{image_id}", method: "POST", permission: Permissions::CHANGE_USER_SETTING)) {
+            $image_id = int_escape($event->get_arg('image_id'));
+            $page->set_mode(PageMode::REDIRECT);
+            $page->set_redirect(make_link("set_avatar/$image_id"));
+        }
+
+        if ($event->page_matches("set_avatar/{image_id}", method: "GET", permission: Permissions::CHANGE_USER_SETTING)) {
+            if (!$user->is_anonymous()) {
```

Comment:
Why the extra check for anons? I would think just checking for "change user setting" permission should already exclude anons (unless for some weird reason the admin has decided to give anons permission to edit their settings, but in that case it seems like they should be allowed to)